### PR TITLE
WebSocket support for HttpClientZioBackend

### PR DESCRIPTION
**Pull request by magdzikk**
Before submitting pull request:
- [x] Check if the project compiles by running `sbt compile`
- [x] Verify docs compilation by running `sbt compileDocs`
- [x] Check if tests pass by running `sbt test`
- [x] Format code by running `sbt scalafmt`

### CODE REVIEW

File: httpclient-backend/zio/src/main/scala/sttp/client/httpclient/zio/ZioWebSocketHandler.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import sttp.client.ws.{WebSocket, WebSocketEvent}
+import sttp.client.ws.internal.AsyncQueue
+import sttp.model.ws.WebSocketBufferFull
+import zio.{Queue, Runtime, Task, ZIO}
+
+object ZioWebSocketHandler {
+  private class ZioAsyncQueue[A](queue: Queue[A], runtime: Runtime[Any]) extends AsyncQueue[BlockingTask, A] {
+    override def offer(t: A): Unit = {
+      if (!runtime.unsafeRun(queue.offer(t))) {
+        throw new WebSocketBufferFull()
```
Comment (adamw) [inline httpclient-backend/zio/src/main/scala/sttp/client/httpclient/zio/ZioWebSocketHandler.scala:15]: maybe this should be in fact an `IllegalStateException` or sth similar indicating a bug - we shouldn't ever get into a situation where the buffer is full?

Comment (magdzikk) [reply httpclient-backend/zio/src/main/scala/sttp/client/httpclient/zio/ZioWebSocketHandler.scala:15]: That's true, the name of the exception isn't right

**Comment by adamw** [comment]
Released in 2.2.1, thanks!

**Comment by adamw** [comment]
& fixed in 2.2.2 :)

**Pull request merged**